service/controller/v5/blobclient/blob_client.go

```diff
@@ -0,0 +1,186 @@
+package blobclient
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-07-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	blobFormatString      = `https://%s.blob.core.windows.net`
+	resourceNotFoundError = "NotFound"
+	maxRetriesRequests    = 3
+)
+
+type Config struct {
+	ContainerName         string
+	GroupName             string
+	StorageAccountName    string
+	StorageAccountsClient *storage.AccountsClient
+}
+
+type BlobClient struct {
+	containerName         string
+	groupName             string
+	storageAccountName    string
+	storageAccountsClient *storage.AccountsClient
+
+	// containerURL is configured separately from the default
+	// parameters.
+	containerURL azblob.ContainerURL
+}
+
+func (c *BlobClient) Boot(ctx context.Context) error {
+	var containerURL azblob.ContainerURL
+
+	key, err := c.getAccountPrimaryKey(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	sc, err := azblob.NewSharedKeyCredential(c.storageAccountName, key)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	p := azblob.NewPipeline(sc, azblob.PipelineOptions{})
+	u, _ := url.Parse(fmt.Sprintf(blobFormatString, c.storageAccountName))
+	service := azblob.NewServiceURL(*u, p)
+	containerURL = service.NewContainerURL(c.containerName)
+	_, err = containerURL.GetProperties(ctx, azblob.LeaseAccessConditions{})
+	if IsContainerNotFound(err) {
+		return nil
```

Comment:
What happens when this code returns `nil` and we call `c.containerURL.NewBlockBlobURL(blobName)`?

Reply:
It will fail as containerURL will be nil at that moment. But we don't do that, we just stop reconciling of the resource and waiting for container to be ready

Reply:
But in this case, we can just drop this additional check as on the next line we have the same masked error return. So It will be handled here https://github.com/giantswarm/azure-operator/pull/395/files#diff-0b6d094c4f73237335bfcd04bfb277c5R53

Reply:
Yes, that's fine as well.